primeqa/components/retriever/dense.py

```diff
@@ -144,3 +147,120 @@ def predict(self, input_texts: List[str], *args, **kwargs) -> Any:
             [(result[0], result[-1]) for result in results_per_query]
             for results_per_query in ranking_results.data.values()
         ]
+
+
+@dataclass
+class DPRRetriever(BaseRetriever):
+    """_summary_
+
+    Args:
+        index_root: str
+        index_name: str
+        checkpoint (str, optional): Model to load. Defaults to checkpoint in index configuration.
+        collection (str, optional): collection to load. Defaults to collection in index configuration.
+        max_num_documents (int, optional): Maximum number of retrieved document. Defaults to 5.
+
+    Important:
+    1. Each field has metadata property which can carry additional information for other downstream usages.
+    2. Two special keys (api_support and exclude_from_hash) are defined in "metadata" property.
+        a. api_support (bool, optional): If set to True, that parameter is exposed via service layer. Defaults to False.
+        b. exclude_from_hash (bool,optional): If set to True, that parameter is not considered while building the hash representation for the object. Defaults to False.
+
+    Returns:
+        _type_: _description_
+
+    """
+
+    checkpoint: str = field(
+        default=None,
+        metadata={
+            "name": "Checkpoint",
+            "description": "Path to checkpoint",
+            "api_support": True,
+        },
+    )
+    collection: str = field( # remove
```

Comment:
Why does it have `#remove`? Is this parameter not required?

Reply:
Updated as suggested.